File: client/sdk/python/anthropic/messages.py
"""Anthropic /v1/messages endpoint functions."""

from anthropic import Anthropic


def create_client(api_key, base_url):
    return Anthropic(api_key=api_key, base_url=base_url)


def chat(prompt, model, api_key, base_url, max_tokens=1024):
    client = create_client(api_key=api_key, base_url=base_url)
    response = client.messages.create(
        model=model,
        max_tokens=max_tokens,
        messages=[{"role": "user", "content": prompt}],
    )
    return response.content[0].text


def stream_chat(prompt, model, api_key, base_url, max_tokens=1024):
    client = create_client(api_key=api_key, base_url=base_url)
    with client.messages.stream(
        model=model,
        max_tokens=max_tokens,
        messages=[{"role": "user", "content": prompt}],
    ) as stream:
        yield from stream.text_stream

File: client/sdk/typescript/src/anthropic/messages.ts
import type { ClientConfig } from "../config";

import { createClient, DEFAULT_MAX_TOKENS } from "./shared";

export async function chat(
  config: ClientConfig,
  prompt: string,
  model: string,
  maxTokens = DEFAULT_MAX_TOKENS,
): Promise<string> {
  const client = createClient(config);
  const response = await client.messages.create({
    model,
    max_tokens: maxTokens,
    messages: [{ role: "user", content: prompt }],
  });

  for (const block of response.content) {
    if (block.type === "text" && block.text) {
      return block.text;
    }
  }
  return "";
}

export async function* streamChat(
  config: ClientConfig,
  prompt: string,
  model: string,
  maxTokens = DEFAULT_MAX_TOKENS,
): AsyncIterable<string> {
  const client = createClient(config);
  const stream = client.messages.stream({
    model,
    max_tokens: maxTokens,
    messages: [{ role: "user", content: prompt }],
  });

  for await (const event of stream) {
    if (
      event.type === "content_block_delta" &&
      event.delta.type === "text_delta" &&
      event.delta.text
    ) {
      yield event.delta.text;
    }
  }
}
